folly/io/test/CompressionTest.cpp

```diff
@@ -120,17 +120,58 @@ constexpr size_t dataSizeLog2 = 27;  // 128MiB
 RandomDataHolder randomDataHolder(dataSizeLog2);
 ConstantDataHolder constantDataHolder(dataSizeLog2);
 
+// The intersection of the provided codecs & those that are compiled in.
+static std::vector<CodecType> supportedCodecs(std::vector<CodecType> const& v) {
+  std::vector<CodecType> supported;
+  supported.reserve(v.size());
+
+  std::copy_if(
+      std::begin(v),
+      std::end(v),
+      std::back_inserter(supported),
+      hasCodec);
+
+  supported.shrink_to_fit();
+  return supported;
+}
+
+// All compiled-in compression codecs.
+static std::vector<CodecType> const& availableCodecs() {
+  static std::vector<CodecType> codecs;
+  codecs.reserve(static_cast<size_t>(CodecType::NUM_CODEC_TYPES));
```

Comment:
I'm inclined to keep the `reserve()` call since the size is likely to be almost- or equal to `NUM_CODEC_TYPES` (and never larger). Agreed on `shrink_to_fit()` being overkill.